### PR TITLE
[IA-2135] Add GKE cluster cleanup

### PR DESCRIPTION
**Pull request by gpcarr**
Adds GKE cluster deletion to `HttpGoogleBillingDAO`

I have `scalafmt` configured in IntelliJ on file save so a lot of these changes are just `scalafmt` but I will point out in comments where my actual changes are

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -1,114 +1,149 @@
...
+    : ModuleID = "org.broadinstitute.dsde.workbench" %% "workbench-metrics" % workbenchMetricsV excludeAll (excludeWorkbenchUtil)
+
+  val slick: ModuleID = "com.typesafe.slick" %% "slick" % slickV
+  val hikariCP: ModuleID = "com.typesafe.slick" %% "slick-hikaricp" % slickV
+  val mysql: ModuleID = "mysql" % "mysql-connector-java" % "6.0.6"
+  val liquibase: ModuleID = "org.liquibase" % "liquibase-core" % "3.5.3"
+
+  val googleBilling
+    : ModuleID = "com.google.apis" % "google-api-services-cloudbilling" % "v1-rev16-1.23.0" excludeAll (excludeGuavaJDK5)
+  val googleContainerV1
```
Comment (gpcarr) [inline project/Dependencies.scala:103]: added dependency `googleContainerV1`

File: src/main/scala/org/broadinstitute/dsde/workbench/gpalloc/dao/HttpGoogleBillingDAO.scala [hunk 2]
```diff
@@ -105,57 +137,85 @@ class HttpGoogleBillingDAO(appName: String,
...
   }
 
   def deploymentManager: DeploymentManagerV2Beta = {
-    new DeploymentManagerV2Beta.Builder(httpTransport, jsonFactory, credential).setApplicationName(appName).build()
+    new DeploymentManagerV2Beta.Builder(httpTransport, jsonFactory, credential)
+      .setApplicationName(appName)
+      .build()
   }
 
+  def gke: Container = {
```
Comment (gpcarr) [inline src/main/scala/org/broadinstitute/dsde/workbench/gpalloc/dao/HttpGoogleBillingDAO.scala:214]: added `gke` to `HttpGoogleBillingDAO`

File: src/main/scala/org/broadinstitute/dsde/workbench/gpalloc/dao/HttpGoogleBillingDAO.scala [hunk 3]
```diff
@@ -194,6 +270,7 @@ class HttpGoogleBillingDAO(appName: String,
       _ <- cleanupClusters(projectName)
       _ <- cleanupVMs(projectName)
       _ <- cleanupDisks(projectName)
+      _ <- cleanupGkeClusters(projectName)
```
Comment (gpcarr) [inline src/main/scala/org/broadinstitute/dsde/workbench/gpalloc/dao/HttpGoogleBillingDAO.scala:273]: added `cleanupGkeClusters`

File: src/main/scala/org/broadinstitute/dsde/workbench/gpalloc/dao/HttpGoogleBillingDAO.scala [hunk 4]
```diff
@@ -454,40 +641,88 @@ class HttpGoogleBillingDAO(appName: String,
...
       diskNames = disks.map(i => i.getName)
       _ <- sequentially(diskNames) { diskName =>
-        googleRq(computeManager.disks().delete(projectName, "us-central1-a", diskName))
+        googleRq(
+          computeManager.disks().delete(projectName, "us-central1-a", diskName)
+        )
+      }
+    } yield {}
+
+  def cleanupGkeClusters(projectName: String): Future[Unit] =
```
Comment (gpcarr) [inline src/main/scala/org/broadinstitute/dsde/workbench/gpalloc/dao/HttpGoogleBillingDAO.scala:702]: added method `cleanupGkeClusters`

**Comment by rtitle** [comment]
Maybe we could turn off scalafmt for this PR? Then potentially `scalafmt` the entire repo in a separate PR. Unfortunately it makes things a little hard to review

File: project/Dependencies.scala [hunk 5]
```diff
@@ -1,114 +1,149 @@
-import Dependencies.{excludeFindbugsJsr, excludeHttpComponent, excludeJacksonCore}
+import Dependencies.{
```
Comment (rtitle) [inline project/Dependencies.scala:1]: I think normally we exclude `Dependencies.scala` from scalafmt

**Comment by gpcarr** [comment]
> Maybe we could turn off scalafmt for this PR? Then if we want we could potentially `scalafmt` the entire repo in a separate PR. Unfortunately it makes things a little hard to review.

Not sure I see an option to undo scalafmt. I can set up another branch and PR that without scalafmt or do you know how to undo scalafmt?

**Comment by rtitle** [comment]
Yeah don't know how to `unscalafmt` (would be a good feature request :)). Setting up a new branch/PR might be easiest, sorry.

**Comment by rtitle** [comment]
Closing in favor of https://github.com/broadinstitute/gpalloc/pull/102

**Pull request closed without merging**